### PR TITLE
Feat/cmc 1864

### PR DESCRIPTION
**Pull request by bhssamray**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CMC-1864

### Change description ###
Party references in 3 email templates - part 1

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sankaviv1 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java [hunk 1]
```diff
@@ -169,4 +172,52 @@ void shouldReturnEmpty_whenPartyTypeIsNotIndividualOrSoleTrader(Party.Type type)
...
+        @Test
+        void shouldReturnReferences_whenNoRefsAvailable() {
+            CaseData caseData = CaseDataBuilder.builder().build();
+
+            String partyReferences = PartyUtils.buildPartiesReferences(caseData);
+
+            assertEquals("", partyReferences);
+        }
+
+        @Test
```
Comment (sherlynkhaw) [inline src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java:188]: maybe shouldReturnReferences_whenCaseHasOneDefendantSolicitor()?

File: src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java [hunk 2]
```diff
@@ -169,4 +172,52 @@ void shouldReturnEmpty_whenPartyTypeIsNotIndividualOrSoleTrader(Party.Type type)
...
+        void shouldReturnReferences_whenNot1v2DiffSolicitorAndBothRefAvailable() {
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimDraft().build();
+
+            String partyReferences = PartyUtils.buildPartiesReferences(caseData);
+
+            assertEquals("Claimant reference: 12345\nDefendant reference: 6789", partyReferences);
+        }
+
+        @Test
+        void shouldReturnReferences_whenNot1v2DiffSolicitorAndOnlyClaimantRefAndSol2RefAvailable() {
```
Comment (sherlynkhaw) [inline src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java:198]: when not 1v2 different solicitor, case won't have solicitor 2.
Is this a 1v2DiffSolicitor case?

Comment (bhssamray) [reply src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java:198]: As the name suggests, this is a scenario of 1v2 Different Solictors, but only claimant and defendant sol 2 refs are available. defendant ref 1 is not available. so in the outcome, we'll only expect the entries which are present. (notably all three refs are optional in this scenario, similarly the claimant and defendant 1 refs are optional in 1v1 case or 1v2 same sol case). I hope I understood the comment right.

File: src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java (hunk 1 above)
Comment (bhssamray) [reply src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java:188]: Are you suggesting a new test or renaming this one on line 189 (shouldReturnReferences_whenNot1v2DiffSolicitorAndBothRefAvailable)? The shouldReturnReferences_whenNot1v2DiffSolicitorAndBothRefAvailable test tests possibly 1v1 and both refs available. I named this scenario this way to signify that this works for 1v1, 2v1, 1v2 same solicitor (basically Not 1v2 diff solicitor).

**Review by sankaviv1 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java (hunk 1 above)
Comment (bhssamray) [reply src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java:188]: The name of test has been changed

File: src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java (hunk 2 above)
Comment (bhssamray) [reply src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java]: The name of test has been corrected

**Review by sherlynkhaw — DISMISSED** [review]

**Review by douglasrice — APPROVED** [review]

**Pull request merged**